index.js

```diff
@@ -174,7 +174,16 @@ InfluxDB.prototype.getDatabaseNames = function (callback) {
     if (err) {
       return callback(err, results)
     }
-    return callback(err, _.map(results[0].series[0].values, function (dbarray) { return dbarray[0] }))
+
+    if (_.isArray(results) && !_.isEmpty(results) && _.isArray(results[0].series) &&
+        !_.isEmpty(results[0].series) && _.isArray(results[0].series[0].values)) {
+
+      return callback(err, _.map(results[0].series[0].values, function (dbarray) {
+        if (_.isArray(dbarray) && !_.isEmpty(dbarray)) {
+          return dbarray[0]
+        }
+      }))
+    } else { return callback(new Error('bad response from server', results))}
```

Comment:
Seem fair. This lib as it stands right now is a little unstable as far as data handling goes; edge cases sometimes aren't handled, some methods don't work on newer influx versions, etc... If I ever get some free time (maybe possible now that my housing and job situation has settled down!) I'll make a pass through all this code and get things in shape, based on reading of Influx's code and extensive testing of its outputs...

tl;dr sounds good for now! 😄